processor/src/fast/mod.rs

```diff
@@ -320,17 +323,113 @@ impl FastProcessor {
         self.execute_impl(program, host).await
     }
 
-    /// Executes the given program and returns the stack outputs.
-    ///
-    /// This function is mainly split out of `execute()` for testing purposes so that we can access
-    /// the internal state of the `FastProcessor` after execution.
     async fn execute_impl(
         &mut self,
         program: &Program,
         host: &mut impl AsyncHost,
     ) -> Result<StackOutputs, ExecutionError> {
-        self.execute_mast_node(program.entrypoint(), program.mast_forest(), program.kernel(), host)
-            .await?;
+        let mut continuation_stack = ContinuationStack::new(program);
+
+        while let Some((current_forest, processing_step)) = continuation_stack.pop_continuation() {
+            match processing_step {
+                Continuation::ExecuteBasicBlock(node_id) => {
+                    let basic_block_node = current_forest[node_id].unwrap_basic_block();
+                    self.execute_basic_block_node(basic_block_node, node_id, &current_forest, host)
+                        .await?
+                },
+                Continuation::StartJoin(node_id) => {
+                    let join_node = current_forest[node_id].unwrap_join();
+                    self.start_join_node(
+                        join_node,
+                        node_id,
+                        &current_forest,
+                        &mut continuation_stack,
+                        host,
+                    )?
```

Comment:
nit: we could look up the `join_node` inside the `start_join_node()` function. Then, this would look something like:
```rust
self.start_join_node(node_id, &current_forest, &mut continuation_stack, host)?
```
(i.e., passing both `join_node` and `node_id` into the function is a bit redundant)

This also applies to other node types.

Reply:
Ended up reverting this change due to your [other suggestion](https://github.com/0xMiden/miden-vm/pull/1990#discussion_r2211939045) for most nodes (all except `Dyn` and `External` which there were borrow-checker issues), since it made reduced the number of `MastForest` accesses.